api/loader/fileloader.go

```diff
@@ -13,11 +12,36 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/pkg/errors"
+
 	"sigs.k8s.io/kustomize/api/ifc"
 	"sigs.k8s.io/kustomize/api/internal/git"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
 )
 
+// DemandDirectoryRoot assures that the given path is in fact a directory.
+func DemandDirectoryRoot(
```

Comment:
I see what you mean now from our earlier conversation.

I agree that if we are going to make this public, it belongs in `kyaml/filesys`. I suggest adding it to `kyaml/filesys/util.go` (and move its tests accordingly)

Reply:
Sounds good! I originally made this change because I needed `DemandDirectoryRoot` for some argument processing. I've since decided to move this logic to a future PR. Will revert `DemandDirectoryRoot` to its private form and follow your suggestion when (if) I use it.

Reply:
I'm leaning towards adding the function as a method to the `FileSystem` interface under `kyaml/filesys/filesystem.go` because it checks that the directory exists on the file system. The functions in `kyaml/filesys/util.go` don't require a file system argument; they work purely with paths.

Reply:
#4677 addresses this